pkg/og/convert/pprof/strprofile/profile.go

```diff
@@ -0,0 +1,192 @@
+package strprofile
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	profilev1 "github.com/grafana/pyroscope/api/gen/proto/go/google/v1"
+)
+
+// Options controls what information is included in the stringified output
+type Options struct {
+	NoPrettyPrint bool
+	NoDuration    bool
+	NoTime        bool
+}
+
+// Location represents a denormalized location
+type Location struct {
+	Address string   `json:"address,omitempty"`
+	Lines   []Line   `json:"lines,omitempty"`
+	Mapping *Mapping `json:"mapping,omitempty"`
+}
+
+type Line struct {
+	Function *Function `json:"function"`
+	Line     int64     `json:"line,omitempty"`
+}
+
+type Function struct {
+	Name       string `json:"name"`
+	SystemName string `json:"system_name,omitempty"`
+	Filename   string `json:"filename,omitempty"`
+	StartLine  int64  `json:"start_line,omitempty"`
+}
+
+type Mapping struct {
+	Start    string `json:"start"`
+	Limit    string `json:"limit"`
+	Offset   string `json:"offset,omitempty"`
+	Filename string `json:"filename,omitempty"`
+	BuildID  string `json:"build_id,omitempty"`
+}
+
+type SampleType struct {
+	Type string `json:"type"`
+	Unit string `json:"unit"`
+}
+
+type Label struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
+type Sample struct {
+	Locations []Location `json:"locations,omitempty"`
+	Values    []int64    `json:"values"`
+	Labels    []Label    `json:"labels,omitempty"`
+}
+
+type Profile struct {
+	SampleTypes   []SampleType `json:"sample_types"`
+	Samples       []Sample     `json:"samples"`
+	TimeNanos     string       `json:"time_nanos,omitempty"`
+	DurationNanos string       `json:"duration_nanos,omitempty"`
+	Period        string       `json:"period,omitempty"`
+}
+
+// Stringify converts a profile to a human-readable JSON representation
+func Stringify(p *profilev1.Profile, opts Options) (string, error) {
```

Comment:
Not really a suggestion/nit, just wondering what you think about that other format.

Not too sure if you considered this: https://github.com/google/pprof/blob/40e02aabc2ad57b2edbe892564a0730aa40b1e50/profile/profile.go#L553-L555

That's my goto, as I fairly familar with the output from `pprof -raw`

I guess you would need to convert to protobuf and parse it to a googleprofile.Profile.

Reply:
I think google profile string is harder to read and write by hand or code.
I also would like to have an option to not output function/location/mapping ids.
Although I like how google string is more compact that what I submit in this PR, I will try update the PR to make it more compact

Another "problem" is the "Time" field which is generated by pyroscope at query time. Although it is pretty easy to remove it from the data.